sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/EventingTestKit.java

```diff
@@ -0,0 +1,176 @@
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  static EventingTestKit start(ActorSystem system, String host, int port, MessageCodec codec) {
+    return EventingTestKitImpl.start(system, host, port, codec);
+  }
+
+
+  Topic getTopic(String topic);
+
+  /**
+   * Testkit utility to mock broker's topic. Useful when doing integration tests for services that do eventing (in or out) to a broker's topic.
+   */
+  @ApiMayChange
+  interface Topic {
+
+    /**
+     * Waits for predefined amount of time (see {@link TopicImpl$#DefaultTimeout()} for default value). If a message arrives in the meantime or
+     * has arrived before but was not consumed, the test fails.
+     */
+    void expectNone();
+
+    /**
+     * Waits for given amount of time. If a message arrives in the meantime or
+     * has arrived before but was not consumed, the test fails.
+     *
+     * @param timeout amount of time to wait for a message
+     */
+    void expectNone(Duration timeout);
+
+    /**
+     * Waits and returns the next unread message on this topic. Note the message might have been received before this
+     * method was called. If no message is received, a timeout exception is thrown.
+     *
+     * @return a Message with a ByteString payload
+     */
+    Message<ByteString> expectOneRaw();
+
+    /**
+     * Waits and returns the next unread message on this topic. Note the message might have been received before this
+     * method was called. If no message is received, a timeout exception is thrown.
+     *
+     * @param timeout amount of time to wait for a message
+     * @return a Message with a ByteString payload
+     */
+    Message<ByteString> expectOneRaw(Duration timeout);
+
+    /**
+     * Waits for predefined amount of time (see {@link TopicImpl$#DefaultTimeout()} for default value) and returns the next unread message on this topic.
+     * Note the message might have been received before this method was called.
+     * If no message is received, a timeout exception is thrown.
+     *
+     * @return message including ByteString payload and metadata
+     */
+    Message<?> expectOne();
+
+    /**
+     * Waits for a specific amount and returns the next unread message on this topic.
+     * Note the message might have been received before this method was called.
+     * If no message is received, a timeout exception is thrown.
+     *
+     * @param timeout amount of time to wait for a message if it was not received already
+     * @return message including ByteString payload and metadata
+     */
+    Message<?> expectOne(Duration timeout);
+
+    /**
+     * Waits and returns the next unread message on this topic and automatically parses
+     * and casts it to the specified given type.
+     *
+     * @param instance class type to cast the received message bytes to
+     * @return a Message of type T
+     * @param <T> a given domain type
+     */
+    <T extends GeneratedMessageV3> Message<T> expectOneTyped(Class<T> instance);
```

Comment:
For protobuf sdks that will differ from how they deal with JSON in actual code, so when we discussed this earlier we said it should reflect the manual JsonSupport decoding instead of doing magic here.

Reply:
I opened this side PR https://github.com/lightbend/kalix-jvm-sdk/pull/1663 so we can discuss further and maybe have an idea of how this would work for java sdk. Please have a look 🙇